### PR TITLE
Update mechanism.md

### PR DESCRIPTION
**Pull request by c3rnst**
Included details to the auctioneer powers 
@Velenir

### CODE REVIEW

File: source/mechanism.md [hunk 1]
```diff
@@ -115,14 +115,46 @@ The primary aim with this deployment was to provide the [dxDAO](https://dutchx.r
 
 ### Some more details to the auctioneer powers:
 #### De- & whitelisting of tokens that generate Magnolia
-TODO
+Whitelisted tokens - if traded in a whitelisted pair - generate Magnolia, which can be used to reduce liquditiy contribution.
+Read above about [Magnolia](https://dutchx.readthedocs.io/en/latest/mechanism.html#magnolia), [Whitelist](https://dutchx.readthedocs.io/en/latest/mechanism.html#whitelist) and [Liquidity contribution](https://dutchx.readthedocs.io/en/latest/mechanism.html#liquidity-contribution).<br/>
+- Whitelisting (and de-whitelisting) can happen by token individually or by entire lists
+- It is not necessary that the token is either listed for trading or actively traded
+- The auctioneer calls the function *updateApprovalOfToken* in the [Token Whitelist contract](https://github.com/gnosis/dx-contracts/blob/master/contracts/base/TokenWhitelist.sol#L28)
```
Comment (Velenir) [inline source/mechanism.md]: > in the Token Whitelist contract

and all other `in the * contract`  crate an impression that they are all different contracts when in reality its all DutchExchange that inherits from them.
Better would be

> calls the function *updateApprovalOfToken* on the DutchExchangeProxy contract which is implemented in the [Token Whitelist contract](link)

or

> ... (see implementation in [Contract](link))

File: source/mechanism.md [hunk 2]
```diff
@@ -115,14 +115,46 @@ The primary aim with this deployment was to provide the [dxDAO](https://dutchx.r
...
+- There is no time lag to the execution of this function
+- It applies immediately and is effective upon users *claiming* their tokens of whitelisted tokens (if traded in a whitelisted pair), also if the auction had finished prior to triggering this function
+
 #### Changing the threshold to start auctions
-TODO
+Once this defined sellVolume (deposit) threshold is reached by both opposite auctions (each), and the prior auctions have finished, the new auctions will commence.
```
Comment (Velenir) [inline source/mechanism.md:127]: There's also the case when only one side has reached the threshold and 24hours have passed
https://github.com/gnosis/dx-contracts/blob/master/contracts/DutchExchange.sol#L799
Not sure if this case is significant enough to merit mentioning

File: source/mechanism.md [hunk 3]
```diff
@@ -115,14 +115,46 @@ The primary aim with this deployment was to provide the [dxDAO](https://dutchx.r
...
+- The auctioneer calls the function *updateThresholdNewAuction* of the [Master Contract](https://github.com/gnosis/dx-contracts/blob/master/contracts/DutchExchange.sol) (via the [Proxy](https://github.com/gnosis/dx-contracts/blob/master/contracts/DutchExchangeProxy.sol)
```
Comment (Velenir) [inline source/mechanism.md:128]: need to close the braces `.../DutchExchangeProxy.sol))`

File: source/mechanism.md [hunk 4]
```diff
@@ -115,14 +115,46 @@ The primary aim with this deployment was to provide the [dxDAO](https://dutchx.r
...
+- At the time of writing, this value is set at USD1000 (in uint USD)
+- There is no time lag to calling this function
+- It applies to all auctions (no individual change possible) that have not started (also if currently waiting for funding)
+
 #### Changing the threshold to add tokens to the DutchX protocol
-TODO
+- Once this defined sellVolume (deposit) threshold is reached, the first ever auction for this pair will commence.
+- The very first auction for an entirely new token on the DutchX has to be with WETH, where the defined threshold has to be reached solely on the WETH sellVolume.
+- For the very first aucion of another token pair (without WETH), the defined threshold can be reached combined on both sides of the auctions sellVolumes.
+- The auctioneer calls the function *updateThresholdNewTokenPair* of the [Master Contract](https://github.com/gnosis/dx-contracts/blob/master/contracts/DutchExchange.sol) (via the [Proxy](https://github.com/gnosis/dx-contracts/blob/master/contracts/DutchExchangeProxy.sol)
```
Comment (Velenir) [inline source/mechanism.md:137]: need to close the braces `.../DutchExchangeProxy.sol))`

File: source/mechanism.md (hunk 3 above)
Comment (Velenir) [reply source/mechanism.md:128]: Also here you nicely wrote 
> calls the function * of the [Master Contract] via the Proxy

This is the same situation as above with
> The auctioneer calls the function *updateApprovalOfToken*

File: source/mechanism.md [hunk 5]
```diff
@@ -115,14 +115,46 @@ The primary aim with this deployment was to provide the [dxDAO](https://dutchx.r
...
+- It is not necessary that the token is either listed for trading or actively traded
+- The auctioneer calls the function *updateApprovalOfToken* in the [Token Whitelist contract](https://github.com/gnosis/dx-contracts/blob/master/contracts/base/TokenWhitelist.sol#L28)
+- There is no time lag to the execution of this function
+- It applies immediately and is effective upon users *claiming* their tokens of whitelisted tokens (if traded in a whitelisted pair), also if the auction had finished prior to triggering this function
+
 #### Changing the threshold to start auctions
-TODO
+Once this defined sellVolume (deposit) threshold is reached by both opposite auctions (each), and the prior auctions have finished, the new auctions will commence.
+- The auctioneer calls the function *updateThresholdNewAuction* of the [Master Contract](https://github.com/gnosis/dx-contracts/blob/master/contracts/DutchExchange.sol) (via the [Proxy](https://github.com/gnosis/dx-contracts/blob/master/contracts/DutchExchangeProxy.sol)
+- At the time of writing, this value is set at USD1000 (in uint USD)
```
Comment (Velenir) [inline source/mechanism.md:129]: > (in uint USD)

I feel is unclear.
Maybe `In the contract 1 USD is represented as 1e18 uint256. E.g. USD1000 = 1000000000000000000000`

`uint` is an alias for the more explicit `uint256` in solidity, explicit is better imho

File: source/mechanism.md [hunk 6]
```diff
@@ -115,14 +115,46 @@ The primary aim with this deployment was to provide the [dxDAO](https://dutchx.r
...
+- At the time of writing this value is set at USD1000 (in uint USD)
+- There is no time lag to calling this function
+- It applies to all auctions (no individual change possible) that have not started (also if currently waiting for funding)
+
 #### Setting a new external ETH/USD price feed 
-TODO*(change only executed with a 30day time lag)*
+The DutchX needs an ETH-USD price feed for two purposes: (i) to calculate the thresholds mentioned above as well as (ii) calculating the liquidity contribution which can be settled in USD. 
+- The auctioneer calls the function *initiateEthUsdOracleUpdate* in the [EthOracle Contract](https://github.com/gnosis/dx-contracts/blob/master/contracts/base/EthOracle.sol#L20)
+- There is a 30day time lag for execution of this function 
+- After the 30days are up, the auctioneer calls the function *updateEthUSDOracle* in the same contract
```
Comment (Velenir) [inline source/mechanism.md:146]: In the next sentence you correctly say that anyone can call `updateEthUSDOracle`
Maybe change this one to `the auctioneer or anyone else`

File: source/mechanism.md [hunk 7]
```diff
@@ -115,14 +115,46 @@ The primary aim with this deployment was to provide the [dxDAO](https://dutchx.r
...
+- A slightly more technical side-note: though  *updateETHUSDOracle* may be triggerd by anyone, only the auctioneer can commence the 30 day time period.
+
 #### Updating the DutchX contract logic 
-TODO*(change only executed with a 30day time lag)*.
+Due to the proxy architecture of the smart contracts, the auctioneer may update the entire DutchX Mastercontract, hence being able to change anything part of the DutchX logic (not only the here mentioned modifiable parameters), *likely* without the need for integrations to be renewed and deposits to be withdrawn.
```
Comment (Velenir) [inline source/mechanism.md:150]: > change any~thing~ part of the DutchX logic (not only the ~here mentioned~ modifiable parameters _**mentioned here**_)
?

File: source/mechanism.md (hunk 1 above)
Comment (Velenir) [reply source/mechanism.md]: or like your
> the function * of the [Implementation Contract] (via the [Proxy])

below

File: source/mechanism.md [hunk 8]
```diff
@@ -115,14 +115,46 @@ The primary aim with this deployment was to provide the [dxDAO](https://dutchx.r
...
+- There is no time lag to calling this function
+- It applies to all auctions (no individual change possible) that have not started (also if currently waiting for funding)
+
 #### Changing the threshold to add tokens to the DutchX protocol
-TODO
+- Once this defined sellVolume (deposit) threshold is reached, the first ever auction for this pair will commence.
+- The very first auction for an entirely new token on the DutchX has to be with WETH, where the defined threshold has to be reached solely on the WETH sellVolume.
+- For the very first aucion of another token pair (without WETH), the defined threshold can be reached combined on both sides of the auctions sellVolumes.
+- The auctioneer calls the function *updateThresholdNewTokenPair* of the [Master Contract](https://github.com/gnosis/dx-contracts/blob/master/contracts/DutchExchange.sol) (via the [Proxy](https://github.com/gnosis/dx-contracts/blob/master/contracts/DutchExchangeProxy.sol)
+- At the time of writing this value is set at USD1000 (in uint USD)
```
Comment (Velenir) [inline source/mechanism.md:138]: abovementioned `(in uint USD)`

**Review by Velenir — CHANGES_REQUESTED** [review]

File: source/mechanism.md (hunk 1 above)
Comment (c3rnst) [inline source/mechanism.md]:
```suggestion
- The auctioneer calls the function *updateApprovalOfToken* on the DutchExchangeProxy contract which is implemented in the [Token Whitelist contract](https://github.com/gnosis/dx-contracts/blob/master/contracts/base/TokenWhitelist.sol#L28)
```

File: source/mechanism.md (hunk 2 above)
Comment (c3rnst) [inline source/mechanism.md:127]:
```suggestion
Once this defined sellVolume (deposit) threshold is reached by both opposite auctions (each), and the prior auctions have finished, the new auctions will commence. *(Technical note: only one side has to reach the threshold if after 24h the thresholds have not been reached)*
```

File: source/mechanism.md (hunk 3 above)
Comment (c3rnst) [inline source/mechanism.md:128]:
```suggestion
- The auctioneer calls the function *updateThresholdNewAuction* of the [Master Contract](https://github.com/gnosis/dx-contracts/blob/master/contracts/DutchExchange.sol) (via the [Proxy](https://github.com/gnosis/dx-contracts/blob/master/contracts/DutchExchangeProxy.sol))
```

File: source/mechanism.md (hunk 5 above)
Comment (c3rnst) [reply source/mechanism.md:129]:
```suggestion
- At the time of writing, this value is set at USD1000 *(Technical note: In the contract 1 USD is represented as 1e18 uint256. E.g. USD1000 = 1000000000000000000000)*
```

File: source/mechanism.md (hunk 4 above)
Comment (c3rnst) [inline source/mechanism.md:137]:
```suggestion
- The auctioneer calls the function *updateThresholdNewTokenPair* of the [Master Contract](https://github.com/gnosis/dx-contracts/blob/master/contracts/DutchExchange.sol) (via the [Proxy](https://github.com/gnosis/dx-contracts/blob/master/contracts/DutchExchangeProxy.sol))
```

File: source/mechanism.md (hunk 8 above)
Comment (c3rnst) [inline source/mechanism.md:138]:
```suggestion
- At the time of writing this value is set at USD1000 *(Technical note: In the contract 1 USD is represented as 1e18 uint256. E.g. USD1000 = 1000000000000000000000)*
```

File: source/mechanism.md (hunk 6 above)
Comment (c3rnst) [inline source/mechanism.md:146]:
```suggestion
- After the 30days are up, the auctioneer or anyone else calls the function *updateEthUSDOracle* in the same contract
```

File: source/mechanism.md (hunk 7 above)
Comment (c3rnst) [inline source/mechanism.md:150]:
```suggestion
Due to the proxy architecture of the smart contracts, the auctioneer may update the entire DutchX Mastercontract, hence being able to change any part of the DutchX logic (not only the modifiable parameters mentioned here), *likely* without the need for integrations to be renewed and deposits to be withdrawn.
```

**Pull request merged**